project/FreeGen2.scala

```diff
@@ -425,7 +431,28 @@ class FreeGen2(managed: List[Class[_]], pkg: String, renames: Map[Class[_], Stri
      s"lazy val ${oname}Interpreter: ${opname} ~> Kleisli[M, $sname, *] = new ${oname}Interpreter { }"
    }
 
-
+   def jdbcApiImports = managed.map(ClassTag(_)).flatMap(imports(_)).distinct.sorted.mkString("\n")
+   def algebrasImports = managed.map(_.getSimpleName).map(c => s"import ${pkg}.${c.toLowerCase}.{ ${c}IO, ${c}Op }").mkString("\n")
+   def commonMethods = s"""
+      |  // Some methods are common to all interpreters and can be overridden to change behavior globally.
+      |  def primitive[J, A](f: J => A): Kleisli[M, J, A] = Kleisli { a =>
+      |    // primitive JDBC methods throw exceptions and so do we when reading values
+      |    // so catch any non-fatal exceptions and lift them into the effect
+      |    blocker.blockOn[M, A](try {
+      |      asyncM.delay(f(a))
+      |    } catch {
+      |      case scala.util.control.NonFatal(e) => asyncM.raiseError(e)
+      |    })(contextShiftM)
```

Comment:
Is this right? I don't think it is - `delay` takes a by-name argument, and so `f(a)` is not evaluated inside of a `try`, so the `try` will never catch any errors. Further, `delay` itself catches `NonFatal`, so why do this explicitly? It's very unclear to me

Reply:
It's copypaste from previous implementation - I just splitted big string into smaller pieces. So, I assume it's correct.